core/src/main/java/org/infinispan/statetransfer/StateTransferManagerImpl.java

```diff
@@ -98,9 +103,11 @@ public void start() throws Exception {
          log.tracef("Starting StateTransferManager of cache %s on node %s", cacheName, rpcManager.getAddress());
       }
 
+      Hash hashFunction = configuration.clustering().hash().hash();
+      componentRegistry.registerComponent(hashFunction, Hash.class);
```

Comment:
But that would require a configuration change (add the attribute) and a bit unclear semantic which has higher priorty; `hash="foo.Bar"` or `hash-factory="foo.BarFactory"`?
Is it really necessary?